addons/escoria-core/game/core-scripts/esc_room.gd

```diff
@@ -123,3 +149,17 @@ func set_camera_limits(p_camera_limits: Array) -> void:
 func set_editor_debug_mode(p_editor_debug_mode: int) -> void:
 	editor_debug_mode = p_editor_debug_mode
 	update()
+
+
+# Runs the script event from the script attached, if any
+#
+# #### Parameters
+#
+# - event_name: the name of the event to run
+func run_script_event(event_name: String):
+	if !esc_script:
+		return
+	var script = escoria.esc_compiler.load_esc_file(esc_script)
```

Comment:
This should be stored in an instance variable so that the script isn't recompiled with every call to run_script_event. Only if that instance variable is empty, the script should be compiled.